airqo-platform/AirQo-api
# set up the pagination support for the GET Events endpoint

# set up the pagination support for the GET Events endpoint

**_WHAT DOES THIS PR DO?_**

- [x] set up the pagination support for the GET Events endpoint
- [x] Refactors many sections of  the `Create Event `use case to accommodate the pagination support

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-865]

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/ft-create-event-PLAT-865

**_HOW DO I TEST OUT THIS PR?_**
README, staging
Please ensure that you update your` .env` accordingly. Just updated the [online version](https://docs.google.com/document/d/1R0ux6i4WaK-NssVQE_9IcTxfUs3ZYIUhSP1Aa7s_hqg/edit?usp=sharing).

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [x] [GET Events](https://app.gitbook.com/o/-MCogOVGQqjUharJKchL/s/-MKqyQkcfs54GYi-q96G/device-registry/events#get-events)
- [x] [Get devices count](https://app.gitbook.com/o/-MCogOVGQqjUharJKchL/s/-MKqyQkcfs54GYi-q96G/device-registry/create-devices#get-the-count-of-devices-on-the-network)

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_SCREENSHOTS?_**
<img width="898" alt="Screenshot 2021-11-29 at 21 41 26" src="https://user-images.githubusercontent.com/1590213/143924628-742f52d1-8d55-43f5-b9e6-9e91ff47fbc2.png">





[PLAT-865]: https://airqoteam.atlassian.net/browse/PLAT-865?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
